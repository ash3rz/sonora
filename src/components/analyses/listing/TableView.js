/**
 * @author sriram
 *
 * A tabular view of analyses
 *
 */
import React from "react";
import {
    build,
    DECheckbox,
    EmptyTable,
    EnhancedTableHead,
    formatDate,
    formatMessage,
    getMessage,
    withI18N,
} from "@cyverse-de/ui-lib";
import {
    IconButton,
    Link,
    makeStyles,
    Paper,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableRow,
    Tooltip,
} from "@material-ui/core";
import {
    HourglassEmptyRounded as HourGlass,
    Launch as LaunchIcon,
    UnfoldMore as UnfoldMoreIcon,
} from "@material-ui/icons";
import ids from "../ids";
import { injectIntl } from "react-intl";
import messages from "../messages";
import constants from "../../../constants";
import analysisStatus from "../../models/analysisStatus";
import TableLoading from "../../utils/TableLoading";

const useStyles = makeStyles((theme) => ({
    name: {
        paddingLeft: theme.spacing(1),
        cursor: "pointer",
    },
}));

function AnalysisName(props) {
    const classes = useStyles();
    const analysis = props.analysis;
    const name = analysis.name;
    const isBatch = analysis.batch;
    const interactiveUrls = analysis.interactive_urls;
    const status = analysis.status;
    const handleInteractiveUrlClick = props.handleInteractiveUrlClick;
    const handleGoToOutputFolder = props.handleGoToOutputFolder;
    const handleBatchIconClick = props.handleBatchIconClick;
    const intl = props.intl;
    const baseId = props.baseId;

    const NameLink = () => {
        return (
            <Tooltip
                id={build(baseId, ids.ANALYSIS_NAME_CELL, ids.TOOLTIP)}
                aria-label={formatMessage(intl, "goOutputFolderOf", {
                    name: name,
                })}
                title={formatMessage(intl, "goOutputFolderOf", { name: name })}
            >
                <Link
                    id={build(baseId, ids.ANALYSIS_NAME_CELL)}
                    onClick={() => handleGoToOutputFolder(analysis)}
                    color="primary"
                    component="button"
                >
                    <span className={classes.name}>{name}</span>
                </Link>
            </Tooltip>
        );
    };

    if (isBatch) {
        return (
            <>
                <NameLink />
                <Tooltip
                    aria-label={formatMessage(intl, "htDetails")}
                    title={getMessage("htDetails")}
                    id={build(baseId, ids.ICONS.BATCH, ids.TOOLTIP)}
                >
                    <IconButton
                        size="small"
                        onClick={() => handleBatchIconClick(analysis)}
                        id={build(baseId, ids.ICONS.BATCH)}
                    >
                        <UnfoldMoreIcon />
                    </IconButton>
                </Tooltip>
            </>
        );
    } else if (
        (status === analysisStatus.SUBMITTED ||
            status === analysisStatus.RUNNING) &&
        interactiveUrls &&
        interactiveUrls.length > 0
    ) {
        return (
            <>
                <NameLink />
                <Tooltip
                    id={build(baseId, ids.ICONS.INTERACTIVE, ids.TOOLTIP)}
                    aria-label={formatMessage(intl, "goToVice")}
                    title={getMessage("goToVice")}
                >
                    <IconButton
                        onClick={() =>
                            handleInteractiveUrlClick(interactiveUrls[0])
                        }
                        size="small"
                        id={build(baseId, ids.ICONS.INTERACTIVE)}
                    >
                        <LaunchIcon />
                    </IconButton>
                </Tooltip>
            </>
        );
    } else {
        return <NameLink />;
    }
}

function AppName(props) {
    const analysis = props.analysis;
    const name = analysis.app_name;
    return <Link component="button">{name}</Link>;
}

function Status(props) {
    const { analysis, username, baseId, analysisUser, intl } = props;
    const allowTimeExtn =
        analysis.interactive_urls &&
        analysis.interactive_urls.length > 0 &&
        analysis.status === analysisStatus.RUNNING &&
        username === analysisUser;
    return (
        <React.Fragment>
            <Link component="button" id={build(baseId, ids.STATUS)}>
                {analysis.status}{" "}
            </Link>
            {allowTimeExtn && (
                <Tooltip
                    aria-label={formatMessage(intl, "extendTime")}
                    title={getMessage("extendTime")}
                    id={build(
                        baseId,
                        ids.BUTTON_EXTEND_TIME_LIMIT,
                        ids.TOOLTIP
                    )}
                >
                    <IconButton
                        id={build(baseId, ids.BUTTON_EXTEND_TIME_LIMIT)}
                        size="small"
                    >
                        <HourGlass />
                    </IconButton>
                </Tooltip>
            )}
        </React.Fragment>
    );
}

const columnData = (intl) => [
    {
        id: ids.NAME,
        name: formatMessage(intl, "name"),
        numeric: false,
        enableSorting: true,
        key: "name",
    },
    {
        id: ids.OWNER,
        name: formatMessage(intl, "owner"),
        numeric: false,
        enableSorting: false,
        key: "owner",
    },
    {
        id: ids.APP,
        name: formatMessage(intl, "app"),
        numeric: false,
        enableSorting: false,
        key: "app",
    },
    {
        id: ids.START_DATE,
        name: formatMessage(intl, "startDate"),
        numeric: false,
        enableSorting: true,
        key: "startdate",
    },
    {
        id: ids.END_DATE,
        name: formatMessage(intl, "endDate"),
        numeric: false,
        enableSorting: true,
        key: "enddate",
    },
    {
        id: ids.STATUS,
        name: formatMessage(intl, "status"),
        numeric: false,
        enableSorting: true,
        key: "status",
    },
];

function TableView(props) {
    const {
        loading,
        error,
        username,
        listing,
        parentId,
        baseId,
        order,
        orderBy,
        selected,
        handleRequestSort,
        handleSelectAllClick,
        handleClick,
        handleInteractiveUrlClick,
        handleGoToOutputFolder,
        handleBatchIconClick,
        intl,
    } = props;

    const analyses = listing?.analyses;
    const tableId = build(baseId, ids.LISTING_TABLE);

    return (
        <TableContainer
            component={Paper}
            style={{
                height: "60vh",
            }}
        >
            <Table
                id={tableId}
                stickyHeader={true}
                size="small"
                aria-label={formatMessage(intl, "ariaTableListing")}
            >
                <EnhancedTableHead
                    baseId={baseId}
                    selectable={true}
                    numSelected={selected.length}
                    rowsInPage={listing?.analyses?.length || 0}
                    order={order}
                    orderBy={orderBy}
                    columnData={columnData(intl)}
                    onRequestSort={handleRequestSort}
                    onSelectAllClick={handleSelectAllClick}
                />
                {loading && (
                    <TableLoading
                        numColumns={columnData(intl).length}
                        numRows={25}
                        baseId={tableId}
                    />
                )}
                {!loading && (
                    <TableBody>
                        {(!analyses || analyses.length === 0) && !error && (
                            <EmptyTable
                                message={getMessage("noAnalyses")}
                                numColumns={columnData(intl).length}
                            />
                        )}
                        {error && (
                            <EmptyTable
                                message={error.toString()}
                                numColumns={columnData(intl).length}
                            />
                        )}
                        {analyses &&
                            analyses.length > 0 &&
                            analyses.map((analysis, index) => {
                                const id = analysis.id;
                                const user =
                                    analysis.username &&
                                    analysis.username.includes(constants.IPLANT)
                                        ? analysis.username.split("@")[0]
                                        : analysis.username;
                                const isSelected = selected.indexOf(id) !== -1;
                                const rowId = build(baseId, tableId, id);
                                return (
                                    <TableRow
                                        onClick={(event) =>
                                            handleClick(event, id, index)
                                        }
                                        role="checkbox"
                                        aria-checked={isSelected}
                                        tabIndex={-1}
                                        selected={isSelected}
                                        hover
                                        key={id}
                                        id={rowId}
                                    >
                                        <TableCell padding="checkbox">
                                            <DECheckbox
                                                id={build(rowId, ids.CHECKBOX)}
                                                checked={isSelected}
                                                tabIndex={0}
                                                inputProps={{
                                                    "aria-label": formatMessage(
                                                        intl,
                                                        "ariaCheckbox",
                                                        {
                                                            label:
                                                                analysis.name,
                                                        }
                                                    ),
                                                }}
                                            />
                                        </TableCell>
                                        <TableCell
                                            id={build(
                                                rowId + ids.ANALYSIS_NAME_CELL
                                            )}
                                        >
                                            <AnalysisName
                                                intl={intl}
                                                analysis={analysis}
                                                baseId={build(
                                                    rowId +
                                                        ids.ANALYSIS_NAME_CELL
                                                )}
                                                parentId={parentId}
                                                handleInteractiveUrlClick={
                                                    handleInteractiveUrlClick
                                                }
                                                handleGoToOutputFolder={
                                                    handleGoToOutputFolder
                                                }
                                                handleBatchIconClick={
                                                    handleBatchIconClick
                                                }
                                            />
                                        </TableCell>
                                        <TableCell>{user}</TableCell>
                                        <TableCell
                                            id={build(rowId, ids.APP_NAME_CELL)}
                                        >
                                            <AppName analysis={analysis} />
                                        </TableCell>
                                        <TableCell>
                                            {formatDate(analysis.startdate)}
                                        </TableCell>
                                        <TableCell>
                                            {formatDate(analysis.enddate)}
                                        </TableCell>
                                        <TableCell
                                            id={build(rowId, ids.SUPPORT_CELL)}
                                        >
                                            <Status
                                                analysis={analysis}
                                                baseId={baseId}
                                                analysisUser={user}
                                                username={username}
                                                intl={intl}
                                            />
                                        </TableCell>
                                    </TableRow>
                                );
                            })}
                    </TableBody>
                )}
            </Table>
        </TableContainer>
    );
}

export default withI18N(injectIntl(TableView), messages);
