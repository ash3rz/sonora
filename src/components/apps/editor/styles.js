export default (theme) => ({
    accordionDetails: {
        flexDirection: "column",
        [theme.breakpoints.down("xs")]: {
            padding: theme.spacing(1),
        },
    },

    deleteIcon: {
        color: theme.palette.error.main,
        backgroundColor: theme.palette.getContrastText(
            theme.palette.error.main
        ),
    },

    formContainer: {
        padding: theme.spacing(1),
    },

    paramCard: {
        margin: theme.spacing(1, 0),
    },
    paramsViewSummary: {
        color: theme.palette.info.contrastText,
        backgroundColor: theme.palette.info.main,
    },
    paramsViewsExpandIcon: {
        color: theme.palette.info.contrastText,
    },
});
