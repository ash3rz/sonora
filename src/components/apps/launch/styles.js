export default (theme) => ({
    accordionDetails: { flexDirection: "column" },

    stepper: {
        padding: theme.spacing(1),
    },
    stepContainer: {
        overflow: "auto",

    },
    spacer: {
        margin: theme.spacing(1),
        [theme.breakpoints.down("xs")]: {
            margin: theme.spacing(0.3),
        },
    },
    appInfoTypography: {
        color: theme.palette.info.main,
        width: "100%",
        margin: theme.spacing(1),
        [theme.breakpoints.down("xs")]: {
            margin: theme.spacing(0.3),
        },
    },
    bottomNavigation: {
        [theme.breakpoints.down("xs")]: {
            margin: theme.spacing(0.3),
        },
        backgroundColor: theme.palette.primary.main,
    },
    bottomNavigationAction: {
        flexGrow: 1,
        color: theme.palette.primary.contrastText,
    },
    paramsReview: {
        margin: theme.spacing(1),
        [theme.breakpoints.down("xs")]: {
            margin: theme.spacing(0.3),
        },
    },
    paramsViewSummary: {
        color: theme.palette.info.contrastText,
        backgroundColor: theme.palette.info.main,
    },
    paramsViewsExpandIcon: {
        color: theme.palette.info.contrastText,
    },
    inputSelectorBrowseButton: {
        marginLeft: theme.spacing(1),
        [theme.breakpoints.down("xs")]: {
            marginLeft: theme.spacing(0.1),
        },
    },
    inputSelectorTextFiled: {
        paddingRight: theme.spacing(1),
        [theme.breakpoints.down("xs")]: {
            paddingRight: theme.spacing(0.1),
        },
    },
});
