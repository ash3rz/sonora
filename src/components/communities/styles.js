const styles = (theme) => ({
    divider: {
        flexGrow: 1,
    },

    filter: {
        width: 200,
        marginRight: theme.spacing(2),
    },
});

export default styles;
