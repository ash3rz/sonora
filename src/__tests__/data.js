import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "@material-ui/styles";
import { DataTableViewTest } from "../../stories/data/TableView.stories";
import { getMuiTheme } from "@cyverse-de/ui-lib";

test("Data Table View renders", () => {
    const component = renderer.create(
        <ThemeProvider theme={getMuiTheme()}>
            <DataTableViewTest />
        </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
