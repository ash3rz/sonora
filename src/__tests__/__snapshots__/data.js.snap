// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table View renders 1`] = `
Array [
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "overflow": "auto",
      }
    }
  >
    <table
      aria-label="/iplant/home/ipcdev Data Table"
      className="MuiTable-root MuiTable-stickyHeader"
      id="tableView.listingTable"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            scope="col"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-label={
                    <FormattedMessage
                      id="selectAll"
                      values={Object {}}
                    />
                  }
                  checked={false}
                  className="PrivateSwitchBase-input-7"
                  data-indeterminate={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            id="tableView.listingTable.checkbox"
            scope="col"
          >
            
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            id="tableView.listingTable.name"
            scope="col"
          >
            <span
              aria-describedby={null}
              aria-disabled={false}
              className="MuiButtonBase-root MuiTableSortLabel-root"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
              title={null}
            >
              Name
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            id="tableView.listingTable.dotMenu"
            scope="col"
          >
            <button
              aria-controls="tableView.listingTable.customCols"
              aria-haspopup="true"
              aria-label="Customize table columns"
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              />
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          aria-checked={false}
          aria-label={null}
          className="MuiTableRow-root MuiTableRow-hover"
          id="tableView.listingTable.analyses_qa-3"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
          title={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-label="analyses_qa-3 checkbox"
                  checked={false}
                  className="PrivateSwitchBase-input-7"
                  data-indeterminate={false}
                  id="tableView.listingTable.analyses_qa-3.checkbox"
                  onChange={[Function]}
                  tabIndex={0}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="makeStyles-dataLink-14"
              id="tableView.listingTable.analyses_qa-3.navigationLink"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              analyses_qa-3
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
          >
            <button
              aria-controls="tableView.listingTable.analyses_qa-3.moreMenu"
              aria-haspopup="true"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              disabled={false}
              id="tableView.listingTable.analyses_qa-3.dotMenuIcon"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <form
              onReset={[Function]}
              onSubmit={[Function]}
            />
          </td>
        </tr>
        <tr
          aria-checked={false}
          aria-label={null}
          className="MuiTableRow-root MuiTableRow-hover"
          id="tableView.listingTable.test"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
          title={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-label="test checkbox"
                  checked={false}
                  className="PrivateSwitchBase-input-7"
                  data-indeterminate={false}
                  id="tableView.listingTable.test.checkbox"
                  onChange={[Function]}
                  tabIndex={0}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="makeStyles-dataLink-14"
              id="tableView.listingTable.test.navigationLink"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              test
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
          >
            <button
              aria-controls="tableView.listingTable.test.moreMenu"
              aria-haspopup="true"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              disabled={false}
              id="tableView.listingTable.test.dotMenuIcon"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <form
              onReset={[Function]}
              onSubmit={[Function]}
            />
          </td>
        </tr>
        <tr
          aria-checked={false}
          aria-label={null}
          className="MuiTableRow-root MuiTableRow-hover"
          id="tableView.listingTable.CORE-8424.txt"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
          title={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-label="CORE-8424.txt checkbox"
                  checked={false}
                  className="PrivateSwitchBase-input-7"
                  data-indeterminate={false}
                  id="tableView.listingTable.CORE-8424.txt.checkbox"
                  onChange={[Function]}
                  tabIndex={0}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="makeStyles-dataLink-14"
              id="tableView.listingTable.CORE-8424.txt.navigationLink"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              CORE-8424.txt
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
          >
            <button
              aria-controls="tableView.listingTable.CORE-8424.txt.moreMenu"
              aria-haspopup="true"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              disabled={false}
              id="tableView.listingTable.CORE-8424.txt.dotMenuIcon"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <form
              onReset={[Function]}
              onSubmit={[Function]}
            />
          </td>
        </tr>
        <tr
          aria-checked={false}
          aria-label={null}
          className="MuiTableRow-root MuiTableRow-hover"
          id="tableView.listingTable.sample.csv"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
          title={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  aria-label="sample.csv checkbox"
                  checked={false}
                  className="PrivateSwitchBase-input-7"
                  data-indeterminate={false}
                  id="tableView.listingTable.sample.csv.checkbox"
                  onChange={[Function]}
                  tabIndex={0}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeSmall"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="makeStyles-dataLink-14"
              id="tableView.listingTable.sample.csv.navigationLink"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              sample.csv
            </span>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
          >
            <button
              aria-controls="tableView.listingTable.sample.csv.moreMenu"
              aria-haspopup="true"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              disabled={false}
              id="tableView.listingTable.sample.csv.dotMenuIcon"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <form
              onReset={[Function]}
              onSubmit={[Function]}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <input
    id="tableView.listingTable.uploadMenuItem.input"
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />,
]
`;
